easybuild/easyconfigs/p/PETSc/PETSc-3.9.1-foss-2018a-downloaded-deps.eb

```diff
@@ -0,0 +1,35 @@
+name = 'PETSc'
+version = '3.9.1'
+versionsuffix = '-downloaded-deps'
+
+homepage = 'http://www.mcs.anl.gov/petsc'
+description = """PETSc, pronounced PET-see (the S is silent), is a suite of data structures and routines for the
+ scalable (parallel) solution of scientific applications modeled by partial differential equations."""
+
+toolchain = {'name': 'foss', 'version': '2018a'}
+toolchainopts = {'usempi': True, 'pic': True}
+
+source_urls = ['http://ftp.mcs.anl.gov/pub/petsc/release-snapshots']
+sources = [SOURCELOWER_TAR_GZ]
+checksums = ['742e838a35d278693e956ed1ca4592c1d663451f6beea0694bf334aeb67681e8']
+
+# (build) dependencies provided through EasyBuild
+builddependencies = [('CMake', '3.11.1')]
+dependencies = [
+    ('HDF5', '1.10.1'),
+    ('Valgrind', '3.13.0'),
+    ('Bison', '3.0.4'),
```

Comment:
Any reference to using `Bison` as a dependency for `PETSc`?

Reply:
I was using a singularity container to install PETSc, it complained about bison not being available. This is because bison is not in the OS by default. It resulted in this error:
```
*******************************************************************************
         UNABLE to CONFIGURE with GIVEN OPTIONS (see configure.log for details):
-------------------------------------------------------------------------------
PTScotch needs bison installed
*******************************************************************************
```
Then including Bison solved my problem, so I think it is better to include it...

Reply:
OK, thanks for clarifying. But, are you sure that `Bison` should be runtime dependency rather than a build dependency? `Bison` usually is a build-only dependency (but it could be different here, of course).

Reply:
Based on what I see in https://github.com/petsc/petsc/blob/master/config/BuildSystem/config/packages/PTScotch.py, it looks like a build-only dependency for `PETSc` as well.

Also, you should include `flex` as well as a build dependency...

Reply:
I agree, thanks for looking into that!